sdk/nodejs/storage/zMixins.ts

```diff
@@ -401,27 +419,68 @@ export class QueueEventSubscription extends appservice.EventSubscription<QueueCo
     }
 }
 
+function merge(trigger: appservice.BindingDefinition, appSettings: pulumi.Input<{ [key: string]: string }>, otherBindings?: appservice.BindingSettings[]) {
+    if (otherBindings) {
+        const bindings = pulumi.all(otherBindings.map(b => b.binding)).apply(bs => [trigger, ...bs]);
+
+        const perFunctionSettings = otherBindings.map(b => b.settings);
+        const settings =  pulumi.all([appSettings, ...perFunctionSettings]).apply(items => items.reduce((a, b) => ({ ...a, ...b }), {}));
+        return { bindings, settings };
+    } else
+    {
+        return { bindings: [trigger], settings: appSettings };
+    }
+}
+
 /**
  * Azure Function triggered by a Storage Queue.
  */
-export class QueueFunction extends appservice.FunctionBase<QueueContext, Buffer, void> {
+export class QueueFunction extends appservice.FunctionBase<QueueContext, Buffer, void | Record<string, any>> {

```

Comment:
What is that `| Record<string, any>`  stuff all about?

Reply:
So that a callback could return arbitrary records with binding data:

``` ts
callback: async (context, msg) => {
    const text = msg.toString();
    return {
        queueOut: `Forwarding ${text}`,
    };
}
```

Reply:
I added an alias for this type with a comment